File: solved/1561.MaximumNumberOfCoinsYouCanGet.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
  int maxCoins(vector<int>& piles) {
    sort(piles.begin(), piles.end());
    int res = 0;
    int minI = 0;
    int maxI = piles.size() -1;
    while (minI < maxI) {
      maxI--;
      res += piles[maxI--];
      minI++;
    }
    cout << res;
    return res;
  }
};

File: solved/1561.MaximumNumberOfCoinsYouCanGet.py
from typing import List

class Solution:
  def maxCoins(self, piles: List[int]) -> int:
    piles.sort()
    res = 0
    minI, maxI = 0, len(piles)-1
    while minI < maxI:
      res += piles[maxI-1]
      maxI -= 2
      minI += 1
    print(res)
    return res
